### PR TITLE
Support use logrus to transmit trace context to log

### PR DESCRIPTION
**Pull request by mrproliu**
See https://github.com/SkyAPM/go2sky/issues/104
Support the `logrus` log framework

### CODE REVIEW

**Comment by arugal** [comment]
Hi @mrproliu you also need to update the [plugin summary](https://github.com/SkyAPM/go2sky-plugins/blob/master/README.md).

File: logrus/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"context"
+	"github.com/sirupsen/logrus"
+	logrusplugin "github.com/SkyAPM/go2sky-plugins/logrus"
+)
+
+func main() {
+	// init format with custom traceId key
+	logrus.SetFormatter(&logrusplugin.WrapFormat{&logrus.JSONFormatter{}, "TID"})
```
Comment (wu-sheng) [inline logrus/README.md:22]: TID? I think we should inject trace context. TID is an old thing.

Comment (wu-sheng) [reply logrus/README.md:22]: My fault, read this, https://github.com/SkyAPM/go2sky/issues/104#issuecomment-841640658

**Comment by wu-sheng** [comment]
> Hi @mrproliu you also need to update the [plugin summary](https://github.com/SkyAPM/go2sky-plugins/blob/master/README.md).

We should separate that list to `trace plugins` and `log plugins`.

**Comment by mrproliu** [comment]
> > Hi @mrproliu you also need to update the [plugin summary](https://github.com/SkyAPM/go2sky-plugins/blob/master/README.md).
> 
> We should separate that list to `trace plugins` and `log plugins`.

Have finished, Please take a look.

**Comment by wu-sheng** [comment]
Update the PR title please.

File: logrus/example_logrus_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	// test format
+	logrus.SetFormatter(Wrap(&logrus.TextFormatter{
+		DisableTimestamp: true,
+	}, "SW_CTX"))
+	logrus.WithContext(context).Info("test2")
+
+	// Output:
+	// {"SW_CTX":"[,,N/A,N/A,-1]","level":"info","msg":"test1"}
+	// level=info msg=test2 SW_CTX="[,,N/A,N/A,-1]"
```
Comment (wu-sheng) [inline logrus/example_logrus_test.go:53]: Let's add the output into the doc too.

Comment (wu-sheng) [reply logrus/example_logrus_test.go:53]: But more like real data.

Comment (mrproliu) [reply logrus/example_logrus_test.go:53]: > Let's add the output into the doc too.

already add the context format at https://github.com/SkyAPM/go2sky-plugins/pull/19/files#diff-f583953a040e7174c73614f1e1e0d8e81695a399c931598d2e28a4b2bb5494cfR22.

Comment (mrproliu) [reply logrus/example_logrus_test.go:53]: > But more like real data.

Golang will check output is the same with stdout, test failure if not the same. Should I need to add some real data to the comment?

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**